mmdet/apis/ote/extension/datasets/mmdataset.py

```diff
@@ -195,3 +200,27 @@ def get_ann_info(self, idx):
         dataset_item = self.ote_dataset[idx]
         labels = self.labels
         return get_annotation_mmdet_format(dataset_item, labels)
+
+    def filter_small_gt(self, item: dict) -> dict:
+        """
+        Function to filter instances in DatasetItem if its width or height is smaller than self.min_size
+
+        :param item: 'data_info' dict that represents the DatasetItem
+        :return dict: the same dict with filtered instances
+        """
+        dataset_item = item['dataset_item']
+        width, height = dataset_item.width, dataset_item.height
+        filtered_anns = []
+
+        for ann in dataset_item.get_annotations():
+            box = ann.shape
+            if min(box.width * width, box.height * height) >= self.min_size:
+                filtered_anns.append(ann)
+        if len(filtered_anns) == 0:
+            print(f'All instances on the image are smaller than min_size={self.min_size} - the image was skipped')
+            annotations = NullAnnotationSceneEntity()
+        else:
+            annotations = AnnotationSceneEntity(annotations=filtered_anns, kind=AnnotationSceneKind.ANNOTATION)
+
+        item['dataset_item'] = DatasetItemEntity(media=dataset_item.media, annotation_scene=annotations)
```

Comment:
I believe neither dataset item, nor annotation scene can be recreated here. I suggest moving this logic to `get_annotation_mmdet_format` function.

Reply:
done